### PR TITLE
[14.0][IMP] stock_orderpoint_purchase_link: Add smart-button on orderpoints

### PR DESCRIPTION
**Pull request by pedrobaeza**
We provide a link from the orderpoint to the involved purchase orders.

FW-port of #1491 

@Tecnativa TT39644

### CODE REVIEW

File: stock_orderpoint_purchase_link/models/stock_warehouse_orderpoint.py [hunk 1]
```diff
@@ -13,3 +14,17 @@ class StockWarehouseOrderpoint(models.Model):
...
+
+    @api.depends("purchase_line_ids")
+    def _compute_purchase_count(self):
+        for record in self:
+            record.purchase_count = len(record.purchase_line_ids.order_id)
+
+    def button_view_purchases(self):
+        self.ensure_one()
+        orders = self.purchase_line_ids.order_id
+        action = self.env.ref("purchase.purchase_rfq").sudo().read()[0]
```
Comment (rousseldenis) [inline stock_orderpoint_purchase_link/models/stock_warehouse_orderpoint.py]: Please use _for_xml_id

**Review by rousseldenis — CHANGES_REQUESTED** [review]

File: stock_orderpoint_purchase_link/models/stock_warehouse_orderpoint.py (hunk 1 above)
Comment (pedrobaeza) [reply stock_orderpoint_purchase_link/models/stock_warehouse_orderpoint.py]: Well, in this case it's the same, isn't it?

Comment (rousseldenis) [reply stock_orderpoint_purchase_link/models/stock_warehouse_orderpoint.py]: Odoo provides function to do that, so if the implementation changes, nothing to do here...

Comment (pedrobaeza) [reply stock_orderpoint_purchase_link/models/stock_warehouse_orderpoint.py]: OK, changed.

**Review by victoralmau — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**